internal/extproc/translator/openai_gcpvertexai.go

```diff
@@ -46,10 +57,92 @@ func (o *openAIToGCPVertexAITranslatorV1ChatCompletion) ResponseHeaders(headers
 
 // ResponseBody implements [Translator.ResponseBody] for GCP Gemini.
 // This method translates a GCP Gemini API response to the OpenAI ChatCompletion format.
-func (o *openAIToGCPVertexAITranslatorV1ChatCompletion) ResponseBody(respHeaders map[string]string, body io.Reader, endOfStream bool) (
+func (o *openAIToGCPVertexAITranslatorV1ChatCompletion) ResponseBody(respHeaders map[string]string, body io.Reader, _ bool) (
 	headerMutation *extprocv3.HeaderMutation, bodyMutation *extprocv3.BodyMutation, tokenUsage LLMTokenUsage, err error,
 ) {
-	// TODO: Implement response body translation from GCP Gemini to OpenAI format.
-	_, _, _ = respHeaders, body, endOfStream
-	return nil, nil, LLMTokenUsage{}, nil
+	if statusStr, ok := respHeaders[statusHeaderName]; ok {
+		var status int
+		if status, err = strconv.Atoi(statusStr); err == nil {
+			if !isGoodStatusCode(status) {
+				// TODO: Parse GCP error response and convert to OpenAI error format.
+				// For now, just return error response as-is.
```

Comment:
Can we prioritize this TODO, i think it is important translation to deliver user error response.